examples/api/lib/foundation/change_notifier/change_notifier.0.dart

```diff
@@ -0,0 +1,154 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+// Flutter code sample for ChangeNotifier with an AnimatedBuilder
+
+import 'package:flutter/material.dart';
+
+void main() {
+  runApp(const CounterApp());
+}
+
+/// This class holds the counter value and the text color.
+class Counter extends ChangeNotifier {
+  int _counter;
+  Color _textColor;
+
+  /// Creates a [Counter] class and, by default, sets [value] to `0` and
+  /// [textColor] to [Colors.black].
+  Counter({
+    int value = 0,
+    Color textColor = Colors.black,
+  }) : _counter = value, _textColor = textColor;
+
+  /// The counter value.
+  int get value => _counter;
+  set value(int value) {
+    if (_counter != value) {
+      _counter = value;
+      notifyListeners();
+    }
+  }
+
+  /// The text color.
+  Color get textColor => _textColor;
+  set textColor(Color colorValue) {
+    if (_textColor != colorValue) {
+      _textColor = colorValue;
+      notifyListeners();
+    }
+  }
+}
+
+/// An [InheritedWidget] that exposes a [Counter] class to the subtree.
+class CounterWidget extends InheritedWidget {
```

Comment:
maybe: CounterProvider instead of CounterWidget since it doesn't display a Counter.